skeleton fails to compile with clang ToT

**Describe the bug**
Attempting to build the skeleton results in a link error.

**To Reproduce**
1. git clone https://github.com/llvm-mirror/llvm.git @ a10a70238ace1093cad3adeb94814b422bd1b5c1
2. cd llvm/tools
3. git clone https://github.com/llvm-mirror/clang.git @ 6ed0749151866894a67a3e7eefdc1f3a547daa0e
3. cd clang/tools
4. git clone https://github.com/llvm-mirror/clang-tools-extra.git @ d0fd727a5aa3776e4d49061de933772cf7d26c4c
5. Build clang and install to local dir
6. git clone https://github.com/bloomberg/clangmetatool.git @ 398a4788e8e6ede5fe1f6800310fe4510a46e7c0
7. Build clangmetatool and install to local dir (same as clang)
8. Copy skeleton to new git repository and change tool name in CMakeLists.txt
9. Configure with CMake; no problems
10. Build and get link error

```
$ make -j40
Scanning dependencies of target mock-api-tool
[ 50%] Building CXX object CMakeFiles/mock-api-tool.dir/src/main.cpp.o
[100%] Linking CXX executable mock-api-tool
CMakeFiles/mock-api-tool.dir/src/main.cpp.o:(.rodata._ZTIN13clangmetatool15MetaToolFactoryINS_8MetaToolI6MyToolEEEE[_ZTIN13clangmetatool15MetaToolFactoryINS_8MetaToolI6MyToolEEEE]+0x10): undefined reference to `typeinfo for clang::tooling::FrontendActionFactory'
CMakeFiles/mock-api-tool.dir/src/main.cpp.o:(.rodata._ZTIN13clangmetatool8MetaToolI6MyToolEE[_ZTIN13clangmetatool8MetaToolI6MyToolEE]+0x10): undefined reference to `typeinfo for clang::ASTFrontendAction'
collect2: error: ld returned 1 exit status
CMakeFiles/mock-api-tool.dir/build.make:122: recipe for target 'mock-api-tool' failed
make[2]: *** [mock-api-tool] Error 1
CMakeFiles/Makefile2:131: recipe for target 'CMakeFiles/mock-api-tool.dir/all' failed
make[1]: *** [CMakeFiles/mock-api-tool.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

**Expected behavior**
Skeleton should compile and link without errors before any original modifications are made.

**Environment (please complete the following information):**
 - Operating System and Version: 4.4.0-124-generic #148-Ubuntu SMP Wed May 2 13:00:18 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- g++ --version reports g++ (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,6 +38,12 @@ add_library(
   src/propagation/types/value_context_ordering.cpp
   src/propagation/util/get_stmt_from_cfg_element.cpp
 )
+
+# Use the same flags that LLVM used to compile
+list(APPEND CMAKE_MODULE_PATH "${LLVM_CMAKE_DIR}")
+include(AddLLVM)
+llvm_update_compile_flags(clangmetatool)
+
 target_include_directories(
   clangmetatool
   PUBLIC

